File: languages/go/entity/identity/credential_verification.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type CredentialVerification struct {
	Timestamp  int64
	Issuer     string
	Revocation int64
}

func NewCredentialVerificationFromProto(s *proto.CredentialVerification) CredentialVerification {
	if s == nil {
		return CredentialVerification{}
	}
	return CredentialVerification{
		Timestamp:  s.Timestamp,
		Issuer:     s.Issuer,
		Revocation: s.Revocation,
	}
}

func (c CredentialVerification) ToProto() *proto.CredentialVerification {
	return &proto.CredentialVerification{
		Timestamp:  c.Timestamp,
		Issuer:     c.Issuer,
		Revocation: c.Revocation,
	}
}

File: languages/js/src/entity/identity/credential_verification.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities";

export class CredentialVerification {
  timestamp: number;
  issuer: string;
  revocation: number;

  constructor(timestamp: number, issuer: string, revocation: number) {
    this.timestamp = timestamp;
    this.issuer = issuer;
    this.revocation = revocation;
  }

  public toProto(): identityEntitiesProto.CredentialVerification {
    return identityEntitiesProto.CredentialVerification.fromPartial({
      timestamp: this.timestamp,
      issuer: this.issuer,
      revocation: this.revocation
    });
  }

  static fromProto(
    r: identityEntitiesProto.CredentialVerification
  ): CredentialVerification {
    return new CredentialVerification(r.timestamp, r.issuer, r.revocation);
  }
}
